scripts/coe_stats_gui.lua

```diff
@@ -0,0 +1,90 @@
+---@class coe.StatsGUI
+local StatsGUI = {}
+
+local mod_gui = require( "mod-gui" )
+
+-- =============================================================================
+
+local function build_stats_info_frame( statistics_frame )
+  local info_frame = statistics_frame.add{
+    type = "frame",
+    name = "coe_stats_info_frame",
+    direction = "vertical"
+  }
+
+  info_frame.add{
+    type = "label",
+    caption = {"coe-stats-gui.label_launches_title"}
+  }.parent.add{
+    type = "flow",
+    direction = "horizontal"
+  }.add{
+    type = "label",
+    caption = {"coe-stats-gui.label_launches_required"}
+  }.parent.add{
+    type = "label",
+    caption = tostring( global.rocket_silo.required_launches )
+  }.parent.parent.add{
+    type = "flow",
+    direction = "horizontal"
+  }.add{
+    type = "label",
+    caption = {"coe-stats-gui.label_launches_complete"}
+  }.parent.add{
+    type = "label",
+    caption = tostring( global.rocket_silo.total_launches )
+  }
+
+  return info_frame
+end -- build_stats_info_frame
+
+-------------------------------------------------------------------------------
+
+local function open_stats_ui(player)
+  local gui = player.gui.left
+  local statistics_frame =
+      gui.add {
+      type = "frame",
+      name = "coe_statistics_frame",
+      direction = "vertical",
+      caption = {"coe-stats-gui.label_stats_title"}
+  }
+
+  build_stats_info_frame( statistics_frame )
+end -- open_stats_ui
+
+-------------------------------------------------------------------------------
+
+function StatsGUI.onPlayerCreated( event )
+  local player = game.get_player(event.player_index)
+  local dialog = mod_gui.get_button_flow( player )
+  if dialog.coe_button_statistics then dialog.coe_button_statistics.destroy() end
+
+  mod_gui.get_button_flow(player).add({
+    name    = "coe_button_statistics",
+    sprite  = "coe_show_ui",
+    style   = "mod_gui_button",
+    tooltip = {"coe-stats-gui.button_statistics"},
+    type    = "sprite-button"
+  })
+end -- onPlayerCreated
+
+-------------------------------------------------------------------------------
+
+---@param event EventData.on_gui_click
+function StatsGUI.onGuiClick(event)
+  local player = game.get_player(event.player_index)
+
+  if event.element.valid and event.element.name == "coe_button_statistics" then
```

Comment:
event data is guaranteed to be valid (in 99.99%) of the cases. Since no other mods should be messing with our element we can safely omit the .valid

Code Style Check: should use branchless programming whenever possible
```lua
if event.element.name ~= "blah" then return end
```

Reply:
If I click the 'x' on the teleporter UI, I get an exception without this check.

Reply:
Ahhh, I am used to STDLIB where I can return to stop processing an event
```lua
if event.element and event.element.name ~= "blah" then return end
```

Reply:
*more reason to write a simple event lib*